broker/src/main/java/io/zeebe/broker/clustering/atomix/storage/snapshot/AtomixSnapshotStorageService.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.broker.clustering.atomix.storage.snapshot;
+
+import io.atomix.protocols.raft.partition.RaftPartition;
+import io.atomix.storage.journal.JournalReader.Mode;
+import io.zeebe.logstreams.state.SnapshotStorage;
+import io.zeebe.logstreams.storage.atomix.AtomixLogStorageReader;
+import io.zeebe.servicecontainer.Service;
+import io.zeebe.servicecontainer.ServiceStartContext;
+import io.zeebe.servicecontainer.ServiceStopContext;
+
+public class AtomixSnapshotStorageService implements Service<SnapshotStorage> {
+  private static final String RUNTIME_DIRECTORY = "runtime";
+
+  private final RaftPartition partition;
+  private final int maxSnapshotsCount;
+
+  private AtomixSnapshotStorage snapshotStorage;
+
+  public AtomixSnapshotStorageService(final RaftPartition partition, final int maxSnapshotsCount) {
```

Comment:
Why we need an service for this now?

Reply:
The deletion service depends on the snapshot storage, so it was a way to model the dependency and make sure we don't end up closing the storage before the deletion service. I think we could not have it but then the dependency is not really well modeled (at least not with our current way of doing it :sweat_smile: maybe with your PR it doesn't matter)

Reply:
ok fine